src/main/java/fr/insalyon/creatis/moteurlite/GaswMonitor.java

```diff
@@ -0,0 +1,106 @@
+package fr.insalyon.creatis.moteurlite;
+
+import java.net.URI;
+import java.util.List;
+
+import fr.insalyon.creatis.gasw.Gasw;
+import fr.insalyon.creatis.gasw.GaswException;
+import fr.insalyon.creatis.gasw.GaswExitCode;
+import fr.insalyon.creatis.gasw.GaswOutput;
+import fr.insalyon.creatis.gasw.execution.GaswStatus;
+
+/**
+ * 
+ * @author Sandesh Patil [https://github.com/sandepat]
+ * 
+ */
+
+public class GaswMonitor extends Thread {
+    private String workflowId;
+    private String applicationName;
+    private int sizeOfInputs;
+    private Gasw gasw;
+    private WorkflowsDbRepository workflowsDbRepository;
+
+    public GaswMonitor(String workflowId, String applicationName, int sizeOfInputs, Gasw gasw) {
+        this.workflowId = workflowId;
+        this.applicationName = applicationName;
+        this.sizeOfInputs = sizeOfInputs;
+        this.gasw = gasw;
+    }
+
+    @Override
+    public void run() {
+        Integer finishedJobsNumber = 0;
+        Integer successfulJobsNumber = 0;
+        Integer failedJobsNumber = 0;
+        boolean hasSuccessfulJob = false; // Flag to track if at least one job is successful
+        try {
+            workflowsDbRepository.persistProcessors(workflowId, applicationName, sizeOfInputs, successfulJobsNumber, failedJobsNumber);
```

Comment:
replace successfulJobsNumber and failedJobsNumber by 0 and move in the main of MoteurLite (see other comment)